### PR TITLE
Search - only enable queries once tab is active

### PR DESCRIPTION
**Pull request by mozzius**
Keeps track of the active tab, and only enables the queries once the tab is active

## Test plan

Look at the network tab - tabs should only load once you click on them

### CODE REVIEW

File: src/view/screens/Search/Search.tsx [hunk 1]
```diff
@@ -876,6 +914,17 @@ function scrollToTopWeb() {
   }
 }
 
+// takes a bool. returns true once the bool is true, and stays true forever
+function useHasBeenTrue(bool: boolean) {
+  const [isTrue, setIsTrue] = React.useState(bool)
+
+  React.useEffect(() => {
+    if (bool) setIsTrue(true)
```
Comment (gaearon) [inline src/view/screens/Search/Search.tsx:922]: In general pattern like this should be avoided. See here: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

We could fix this, but I wonder if it makes sense to just leave `enabled` on only for the current one? Same approach we do in `Home` and `Profile`. I'd like to at least be consistent with behavior with those — IMO if we change it here, we should do it as a single change across all surfaces (in a separate PR).

Comment (mozzius) [reply src/view/screens/Search/Search.tsx]: Yeah, it's actually fine if the other tabs' queries are disabled when inactive. Removed

**Review by gaearon — APPROVED** [review]
lgtm

**Comment by gaearon** [comment]
needs TS fix

**Pull request merged**